netket/experimental/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,769 @@
+                terms.append(tuple(t))
+                weights.append(w * self._constant)
+        constant = self._constant * other._constant
+
+        self._operators = dict(zip(terms, weights))
+        self._constant = constant
+        return self
+
+    def _op_matmul(self, other):
+        if not isinstance(other, FermionOperator2nd):
+            return NotImplementedError
+        dtype = np.promote_types(self.dtype, other.dtype)
+        op = self.copy(dtype=dtype)
+        op @= other
+        return op
+
+    def __radd__(self, other):
+        return self + other
+
+    def __add__(self, other):
+        dtype = np.promote_types(self.dtype, _dtype(other))
+        op = self.copy(dtype=dtype)
+        op += other
+        return op
+
+    def __iadd__(self, other):
+        if isinstance(other, numbers.Number):
+            if other != 0.0:
+                self._constant += other
+            return self
+        if not isinstance(other, FermionOperator2nd):
+            raise NotImplementedError
+        if not self.hilbert == other.hilbert:
+            raise ValueError(
+                f"Can only add identical hilbert spaces (got A+B, A={self.hilbert}, B={other.hilbert})"
+            )
+        if not np.can_cast(_dtype(other), self.dtype, casting="same_kind"):
+            raise ValueError(
+                f"Cannot add inplace operator with dtype {type(other)} "
+                f"to operator with dtype {self.dtype}"
+            )
+        for t, w in other._operators.items():
+            if t in self._operators.keys():
+                self._operators[t] += w
+            else:
+                self._operators[t] = w
+        self._constant += other._constant
+        return self
+
+    def __sub__(self, other):
+        return self + (-other)
+
+    def __rsub__(self, other):
+        return other + (-self)
+
+    def __neg__(self):
+        return -1 * self
+
+    def __rmul__(self, scalar):
+        return self * scalar
+
+    def __imul__(self, scalar):
+        if not isinstance(scalar, numbers.Number):
+            # we will overload this as matrix multiplication
+            self._iop__matmul__(scalar)
+        if not np.can_cast(_dtype(scalar), self.dtype, casting="same_kind"):
+            raise ValueError(
+                f"Cannot multiply inplace scalar with dtype {type(scalar)} "
+                f"to operator with dtype {self.dtype}"
+            )
+        scalar = np.array(scalar, dtype=self.dtype).item()
+        self._operators = tree_map(lambda x: x * scalar, self._operators)
+        self._constant *= scalar
+        return self
+
+    def __mul__(self, scalar):
+        if not isinstance(scalar, numbers.Number):
+            # we will overload this as matrix multiplication
+            return self._op__matmul__(scalar)
+        dtype = np.promote_types(self.dtype, _dtype(scalar))
+        op = self.copy(dtype=dtype)
+        op *= scalar
+        return op
```

Comment:
we already know that this is good so let's bypass python
```suggestion
        return op.__imul__(scalar)
```